doc/Syscalls.md

```diff
@@ -157,8 +157,9 @@ A callback function is uniquely identified by the pair (`driver`,
 are any pending callbacks for this callback ID, they are removed from the queue
 before the new callback function is bound to the callback ID.
 
-Passing a null pointer callback function disables a previously set callback
-(besides flushing pending callbacks for this callback ID).
+A process can pass a null pointer as the callback argument to request the driver
```

Comment:
I think this is much less clear than the previous documentation. It doesn't specify at all what happens with this "request" to disable the callback. When does the driver take it into account? Can a driver reject such a request?

Without any such guarantees that the callback is indeed disabled once the syscall returns, how can one program anything useful in userspace?